extractor/src/main/java/org/schabi/newpipe/extractor/services/soundcloud/linkHandler/SoundcloudStreamLinkHandlerFactory.java

```diff
@@ -31,6 +32,9 @@ public String getUrl(final String id) throws ParsingException {
 
     @Override
     public String getId(final String url) throws ParsingException {
+        if (Parser.isMatch(API_URL_PATTERN, url)) {
+            return Parser.matchGroup1(API_URL_PATTERN, url);
+        }
```

Comment:
Aren't you parsing the url with the regex twice here? I think you can return  matchGroup1 and sorround it with a try-catch

Reply:
I just benchmarked both approaches, it seems that it does not make a difference.

Time needed for 100000 iterations: 224 ms using current implementation
Time needed for 100000 iterations: 227 ms using try & catch

That was a surprise for me, because throwing exceptions can be a quite heavy task. I guess that's not the case here, because we have a flat hierarchy in my tests.
If you want, I can change the implementation anyway.